src/include/network/routines.rb

```diff
@@ -363,6 +363,12 @@ def ControllerType(hwdevice)
         when 145
           return "usb" # #22739
         when 128
+          # Mellanox ConnectX-3 series badly uses the 128 sub class
+          # Check for the specific known boards with bad subclass.
+          if Ops.get_integer(hwdevice, "vendor_id", 0) == 71_091
+            device_id = Ops.get_integer(hwdevice, "device_id", -1)
+            return "ib" if [69_635, 69_636, 69_639].include?(device_id)
```

Comment:
i don't know why there is an extra bit. This piece of code is dark magic approximately 15 years old. It is being investigated per bug report bases only ;-)